user/views.py

```diff
@@ -0,0 +1,87 @@
+from functools import partial
+from user.serializer import LoginSerializer, ProfileSerializer, UserSerializer
+from rest_framework import authentication, permissions
+from rest_framework.authtoken.models import Token
+from django.contrib.auth import authenticate
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from .utils import admin_required
+from rest_framework import status
+from user.models import Profile, User
+
+# Create your views here.
+
+class Userview(APIView):
+    authentication_classes = [authentication.TokenAuthentication]
+    permission_classes = [permissions.IsAuthenticated, admin_required]
+    serializer_class = UserSerializer
+    
+    
+    
+    def get(self, request, id=None):
+        user = User.objects.all()
+        serializer = self.serializer_class(user, many=True)
+        return Response(serializer.data)
+    
+    def post(self, request):
+        serializer = self.serializer_class(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
```

Comment:
This post method is wrong, why the token isn't created?

Reply:
fixed